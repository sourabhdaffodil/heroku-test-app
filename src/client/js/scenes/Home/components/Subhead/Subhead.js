import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { media } from '../../../../theme/style-utils';
import RestricitedContainer from '../../../../components/Containers/Restricited/Restricited';
import Button from '../../../../components/Button/Button';
import PillLeft from '../../../../components/Button/PillLeft/PillLeft';
import PillRight from '../../../../components/Button/PillRight/PillRight';
import Icon from '../../../../components/Icon/Icon';
import Link from '../../../../components/Link/Link';

const Wrapper = styled.div`
  padding: 30px 0;
  background-color: ${props => props.theme.color('background', 'lighter')};
  
  ${media.phone`
    position: relative;
    padding: 15px 0;
    background-color: ${props => props.theme.color('foreground')};
    color: ${props => props.theme.contrast('dark')};
  `}
`;

const Container = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;

  ${media.phone`
    justify-content: center;
    color: ${props => props.theme.contrast('dark')};
  `}
    
  button {
    ${media.phone`
      display: none;
    `}  
  }
  
  i.fa-book {
    margin-right: 5px;
    color: ${props => props.theme.contrast('light', 'secondary')};
  }

  i.fa-book,
  .link {
    ${props => props.theme.textSetting('md')};

    ${media.phone`
      color: ${props => props.theme.contrast('dark')};
      ${props => props.theme.textSetting('nr')};
    `}
  }  
`;

class Subhead extends Component {
  render() {
    return (
      <Wrapper>
        <RestricitedContainer>
          <Container>
            <div>
              <Icon icon="book" />
              <Link url={this.props.repo} className="link">OrencioRodolfo / <b>orodolfo</b></Link>
            </div>
            <Link url="https://github.com/OrencioRodolfo">
              <Button alt="Github stars" title="Github stars">
                <PillLeft>
                  <Icon icon="star" />
                  Stars
                </PillLeft>
                <PillRight>{this.props.totalStars}</PillRight>
              </Button>
            </Link>
          </Container>
        </RestricitedContainer>
      </Wrapper>
    );
  }
}

Subhead.propTypes = {
  repo: PropTypes.string.isRequired,
  totalStars: PropTypes.number.isRequired,
};

function mapStateToProps(state) {
  return {
    repo: state.about.repository,
    totalStars: state.about.githubStars,
  };
}

export default connect(mapStateToProps)(Subhead);

export {
  Subhead,
  mapStateToProps,
};
