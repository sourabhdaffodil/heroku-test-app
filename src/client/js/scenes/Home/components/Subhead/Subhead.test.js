import React from 'react';
import { shallow } from 'enzyme';
import { Subhead, mapStateToProps } from './Subhead';

describe('Subhead component', () => {
  it('Must render the subhead with no link and zero stars', () => {
    const wrapper = shallow(
      <Subhead repo="" totalStars={0} />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('Must render the subhead with a link and some stars', () => {
    const wrapper = shallow(
      <Subhead repo="sample.com" totalStars={50} />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  describe('Subhead mapStateToProps function', () => {
    it('Must return an object with an empty "repo" string and 0 for "totalStars"', () => {
      const defaultState = {
        about: {
          repository: '',
          githubStars: 0,
        },
      };
      const result = mapStateToProps(defaultState);
      const expected = {
        repo: '',
        totalStars: 0,
      };
      expect(result).toEqual(expected);
    });

    it('Must return an object with an empty "repo" string and 0 for "totalStars"', () => {
      const defaultState = {
        about: {
          repository: 'sample.com',
          githubStars: 10,
        },
      };
      const result = mapStateToProps(defaultState);
      const expected = {
        repo: 'sample.com',
        totalStars: 10,
      };
      expect(result).toEqual(expected);
    });
  });
});
