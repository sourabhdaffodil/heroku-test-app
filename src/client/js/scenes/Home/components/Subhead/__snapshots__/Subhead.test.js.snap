// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subhead component Must render the subhead with a link and some stars 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Subhead
    repo="sample.com"
    totalStars={50}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <Container>
        <styled.div>
          <div>
            <Icon
              icon="book"
            />
            <Link
              className="link"
              url="sample.com"
            >
              OrencioRodolfo / 
              <b>
                orodolfo
              </b>
            </Link>
          </div>
          <Link
            className=""
            url="https://github.com/OrencioRodolfo"
          >
            <Button
              alt="Github stars"
              title="Github stars"
            >
              <PillLeft>
                <Icon
                  icon="star"
                />
                Stars
              </PillLeft>
              <PillRight>
                50
              </PillRight>
            </Button>
          </Link>
        </styled.div>
      </Container>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "children": <styled.div>
          <div>
            <Icon
              icon="book"
            />
            <Link
              className="link"
              url="sample.com"
            >
              OrencioRodolfo / 
              <b>
                orodolfo
              </b>
            </Link>
          </div>
          <Link
            className=""
            url="https://github.com/OrencioRodolfo"
          >
            <Button
              alt="Github stars"
              title="Github stars"
            >
              <PillLeft>
                <Icon
                  icon="star"
                />
                Stars
              </PillLeft>
              <PillRight>
                50
              </PillRight>
            </Button>
          </Link>
        </styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div>
              <Icon
                icon="book"
              />
              <Link
                className="link"
                url="sample.com"
              >
                OrencioRodolfo / 
                <b>
                  orodolfo
                </b>
              </Link>
            </div>,
            <Link
              className=""
              url="https://github.com/OrencioRodolfo"
            >
              <Button
                alt="Github stars"
                title="Github stars"
              >
                <PillLeft>
                  <Icon
                    icon="star"
                  />
                  Stars
                </PillLeft>
                <PillRight>
                  50
                </PillRight>
              </Button>
            </Link>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Icon
                  icon="book"
                />,
                <Link
                  className="link"
                  url="sample.com"
                >
                  OrencioRodolfo / 
                  <b>
                    orodolfo
                  </b>
                </Link>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "icon": "book",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "OrencioRodolfo / ",
                    <b>
                      orodolfo
                    </b>,
                  ],
                  "className": "link",
                  "url": "sample.com",
                },
                "ref": null,
                "rendered": Array [
                  "OrencioRodolfo / ",
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "orodolfo",
                    },
                    "ref": null,
                    "rendered": "orodolfo",
                    "type": "b",
                  },
                ],
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": <Button
                alt="Github stars"
                title="Github stars"
              >
                <PillLeft>
                  <Icon
                    icon="star"
                  />
                  Stars
                </PillLeft>
                <PillRight>
                  50
                </PillRight>
              </Button>,
              "className": "",
              "url": "https://github.com/OrencioRodolfo",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "alt": "Github stars",
                "children": Array [
                  <PillLeft>
                    <Icon
                      icon="star"
                    />
                    Stars
                  </PillLeft>,
                  <PillRight>
                    50
                  </PillRight>,
                ],
                "title": "Github stars",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <Icon
                        icon="star"
                      />,
                      "Stars",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "function",
                      "props": Object {
                        "icon": "star",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "Stars",
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "children": 50,
                  },
                  "ref": null,
                  "rendered": 50,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Container>
          <styled.div>
            <div>
              <Icon
                icon="book"
              />
              <Link
                className="link"
                url="sample.com"
              >
                OrencioRodolfo / 
                <b>
                  orodolfo
                </b>
              </Link>
            </div>
            <Link
              className=""
              url="https://github.com/OrencioRodolfo"
            >
              <Button
                alt="Github stars"
                title="Github stars"
              >
                <PillLeft>
                  <Icon
                    icon="star"
                  />
                  Stars
                </PillLeft>
                <PillRight>
                  50
                </PillRight>
              </Button>
            </Link>
          </styled.div>
        </Container>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": <styled.div>
            <div>
              <Icon
                icon="book"
              />
              <Link
                className="link"
                url="sample.com"
              >
                OrencioRodolfo / 
                <b>
                  orodolfo
                </b>
              </Link>
            </div>
            <Link
              className=""
              url="https://github.com/OrencioRodolfo"
            >
              <Button
                alt="Github stars"
                title="Github stars"
              >
                <PillLeft>
                  <Icon
                    icon="star"
                  />
                  Stars
                </PillLeft>
                <PillRight>
                  50
                </PillRight>
              </Button>
            </Link>
          </styled.div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div>
                <Icon
                  icon="book"
                />
                <Link
                  className="link"
                  url="sample.com"
                >
                  OrencioRodolfo / 
                  <b>
                    orodolfo
                  </b>
                </Link>
              </div>,
              <Link
                className=""
                url="https://github.com/OrencioRodolfo"
              >
                <Button
                  alt="Github stars"
                  title="Github stars"
                >
                  <PillLeft>
                    <Icon
                      icon="star"
                    />
                    Stars
                  </PillLeft>
                  <PillRight>
                    50
                  </PillRight>
                </Button>
              </Link>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Icon
                    icon="book"
                  />,
                  <Link
                    className="link"
                    url="sample.com"
                  >
                    OrencioRodolfo / 
                    <b>
                      orodolfo
                    </b>
                  </Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "icon": "book",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      "OrencioRodolfo / ",
                      <b>
                        orodolfo
                      </b>,
                    ],
                    "className": "link",
                    "url": "sample.com",
                  },
                  "ref": null,
                  "rendered": Array [
                    "OrencioRodolfo / ",
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "orodolfo",
                      },
                      "ref": null,
                      "rendered": "orodolfo",
                      "type": "b",
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "children": <Button
                  alt="Github stars"
                  title="Github stars"
                >
                  <PillLeft>
                    <Icon
                      icon="star"
                    />
                    Stars
                  </PillLeft>
                  <PillRight>
                    50
                  </PillRight>
                </Button>,
                "className": "",
                "url": "https://github.com/OrencioRodolfo",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "alt": "Github stars",
                  "children": Array [
                    <PillLeft>
                      <Icon
                        icon="star"
                      />
                      Stars
                    </PillLeft>,
                    <PillRight>
                      50
                    </PillRight>,
                  ],
                  "title": "Github stars",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        <Icon
                          icon="star"
                        />,
                        "Stars",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "function",
                        "props": Object {
                          "icon": "star",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "Stars",
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "children": 50,
                    },
                    "ref": null,
                    "rendered": 50,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Subhead component Must render the subhead with no link and zero stars 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Subhead
    repo=""
    totalStars={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <Container>
        <styled.div>
          <div>
            <Icon
              icon="book"
            />
            <Link
              className="link"
              url=""
            >
              OrencioRodolfo / 
              <b>
                orodolfo
              </b>
            </Link>
          </div>
          <Link
            className=""
            url="https://github.com/OrencioRodolfo"
          >
            <Button
              alt="Github stars"
              title="Github stars"
            >
              <PillLeft>
                <Icon
                  icon="star"
                />
                Stars
              </PillLeft>
              <PillRight>
                0
              </PillRight>
            </Button>
          </Link>
        </styled.div>
      </Container>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "children": <styled.div>
          <div>
            <Icon
              icon="book"
            />
            <Link
              className="link"
              url=""
            >
              OrencioRodolfo / 
              <b>
                orodolfo
              </b>
            </Link>
          </div>
          <Link
            className=""
            url="https://github.com/OrencioRodolfo"
          >
            <Button
              alt="Github stars"
              title="Github stars"
            >
              <PillLeft>
                <Icon
                  icon="star"
                />
                Stars
              </PillLeft>
              <PillRight>
                0
              </PillRight>
            </Button>
          </Link>
        </styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div>
              <Icon
                icon="book"
              />
              <Link
                className="link"
                url=""
              >
                OrencioRodolfo / 
                <b>
                  orodolfo
                </b>
              </Link>
            </div>,
            <Link
              className=""
              url="https://github.com/OrencioRodolfo"
            >
              <Button
                alt="Github stars"
                title="Github stars"
              >
                <PillLeft>
                  <Icon
                    icon="star"
                  />
                  Stars
                </PillLeft>
                <PillRight>
                  0
                </PillRight>
              </Button>
            </Link>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Icon
                  icon="book"
                />,
                <Link
                  className="link"
                  url=""
                >
                  OrencioRodolfo / 
                  <b>
                    orodolfo
                  </b>
                </Link>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "icon": "book",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "OrencioRodolfo / ",
                    <b>
                      orodolfo
                    </b>,
                  ],
                  "className": "link",
                  "url": "",
                },
                "ref": null,
                "rendered": Array [
                  "OrencioRodolfo / ",
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "orodolfo",
                    },
                    "ref": null,
                    "rendered": "orodolfo",
                    "type": "b",
                  },
                ],
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": <Button
                alt="Github stars"
                title="Github stars"
              >
                <PillLeft>
                  <Icon
                    icon="star"
                  />
                  Stars
                </PillLeft>
                <PillRight>
                  0
                </PillRight>
              </Button>,
              "className": "",
              "url": "https://github.com/OrencioRodolfo",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "alt": "Github stars",
                "children": Array [
                  <PillLeft>
                    <Icon
                      icon="star"
                    />
                    Stars
                  </PillLeft>,
                  <PillRight>
                    0
                  </PillRight>,
                ],
                "title": "Github stars",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <Icon
                        icon="star"
                      />,
                      "Stars",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "function",
                      "props": Object {
                        "icon": "star",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "Stars",
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "children": 0,
                  },
                  "ref": null,
                  "rendered": 0,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Container>
          <styled.div>
            <div>
              <Icon
                icon="book"
              />
              <Link
                className="link"
                url=""
              >
                OrencioRodolfo / 
                <b>
                  orodolfo
                </b>
              </Link>
            </div>
            <Link
              className=""
              url="https://github.com/OrencioRodolfo"
            >
              <Button
                alt="Github stars"
                title="Github stars"
              >
                <PillLeft>
                  <Icon
                    icon="star"
                  />
                  Stars
                </PillLeft>
                <PillRight>
                  0
                </PillRight>
              </Button>
            </Link>
          </styled.div>
        </Container>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": <styled.div>
            <div>
              <Icon
                icon="book"
              />
              <Link
                className="link"
                url=""
              >
                OrencioRodolfo / 
                <b>
                  orodolfo
                </b>
              </Link>
            </div>
            <Link
              className=""
              url="https://github.com/OrencioRodolfo"
            >
              <Button
                alt="Github stars"
                title="Github stars"
              >
                <PillLeft>
                  <Icon
                    icon="star"
                  />
                  Stars
                </PillLeft>
                <PillRight>
                  0
                </PillRight>
              </Button>
            </Link>
          </styled.div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div>
                <Icon
                  icon="book"
                />
                <Link
                  className="link"
                  url=""
                >
                  OrencioRodolfo / 
                  <b>
                    orodolfo
                  </b>
                </Link>
              </div>,
              <Link
                className=""
                url="https://github.com/OrencioRodolfo"
              >
                <Button
                  alt="Github stars"
                  title="Github stars"
                >
                  <PillLeft>
                    <Icon
                      icon="star"
                    />
                    Stars
                  </PillLeft>
                  <PillRight>
                    0
                  </PillRight>
                </Button>
              </Link>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Icon
                    icon="book"
                  />,
                  <Link
                    className="link"
                    url=""
                  >
                    OrencioRodolfo / 
                    <b>
                      orodolfo
                    </b>
                  </Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "icon": "book",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      "OrencioRodolfo / ",
                      <b>
                        orodolfo
                      </b>,
                    ],
                    "className": "link",
                    "url": "",
                  },
                  "ref": null,
                  "rendered": Array [
                    "OrencioRodolfo / ",
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "orodolfo",
                      },
                      "ref": null,
                      "rendered": "orodolfo",
                      "type": "b",
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "children": <Button
                  alt="Github stars"
                  title="Github stars"
                >
                  <PillLeft>
                    <Icon
                      icon="star"
                    />
                    Stars
                  </PillLeft>
                  <PillRight>
                    0
                  </PillRight>
                </Button>,
                "className": "",
                "url": "https://github.com/OrencioRodolfo",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "alt": "Github stars",
                  "children": Array [
                    <PillLeft>
                      <Icon
                        icon="star"
                      />
                      Stars
                    </PillLeft>,
                    <PillRight>
                      0
                    </PillRight>,
                  ],
                  "title": "Github stars",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        <Icon
                          icon="star"
                        />,
                        "Stars",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "function",
                        "props": Object {
                          "icon": "star",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "Stars",
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "children": 0,
                    },
                    "ref": null,
                    "rendered": 0,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
